app/src/Controller/GameController.php

```diff
@@ -5,53 +5,77 @@
 use Symfony\Bundle\FrameworkBundle\Controller\Controller;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Cookie;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
 use App\Api\GameApi;
 use App\Entity\Game;
 use App\Repository\GameRepository;
 
 class GameController extends Controller {
   public const DEFAULT_SIZE = 4;
+  public const COOKIE_NAME = 'current-puzzle';
+
   private $api;
   private $twig;
 
   public function __construct(\Twig_Environment $twig, GameApi $gameApi, GameRepository $gameRepository) {
-    $this->api = $gameApi;
     $this->twig = $twig;
+    $this->api = $gameApi;
     $this->gameRepository = $gameRepository;
   }
 
-  private function renderGrid(string $id, Game $game) {
+  private function hasAccess($request, $game) {
```

Comment:
It's work but it's a little tricky.. Moreover, private functions are difficult to test.

As I said at the office, you should use the Symfony auth / role system (which is heavy for this project)... Or parameter resolving as described bellow.

https://symfony.com/doc/master/controller/argument_value_resolver.html

This system let you create a class / service which is responsible of the parameter resolution of your controller method.

For example, you can trigger it when a "GameContext" object is required in the controller method signature and retrieve the game with the access status from the ID in the URL and the cookie.

Let's discuss it tomorrow ;)